src/Warning/Warning.js

```diff
@@ -0,0 +1,271 @@
+/* @flow */
+
+import * as React from 'react';
+import { Animated, View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { getStatusBarHeight } from 'react-native-status-bar-height';
+
+import { Icon } from '../Icon';
+import Touchable from '../Button/Touchable';
+import StyleSheet from '../PlatformStyleSheet';
+
+import type { StylePropType } from '../PlatformStyleSheet/StyleTypes';
+import type { OnLayout, NotificationType } from '../types';
+
+import Text from '../Text';
+
+type Props = {|
+  +style?: StylePropType,
+  +onDismiss: () => void,
+|};
+
+type State = {|
+  layout: {
+    height: number,
+    measured: boolean,
+  },
+  opacity: Animated.Value,
+  translateY: Animated.Value,
+  isOpen: boolean,
+  warningType: NotificationType,
+  warningTitle: React.Node | string,
+  warningMessage: React.Node | string,
+|};
+
+const ANIMATION_DURATION = 250;
+
+const AnimatedTouchable = Animated.createAnimatedComponent(Touchable);
+
+export default class Warning extends React.Component<Props, State> {
```

Comment:
This component seems the same or 90% the same as `Alert.js`. I think abstraction is required.

Reply:
Will try to extract common codebase or create just one component and distinguish `Alert` and `Warning` using props.